src/app/CommandHandler.cpp

```diff
@@ -247,6 +248,21 @@ CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommand
     VerifyOrExit(mpCallback->CommandExists(ConcreteCommandPath(endpointId, clusterId, commandId)),
                  err = CHIP_ERROR_INVALID_PROFILE_ID);
 
+    {
+        Access::SubjectDescriptor subjectDescriptor; // TODO: get actual subject descriptor
+        Access::RequestPath requestPath{ .cluster = clusterId, .endpoint = endpointId };
+        Access::Privilege requestPrivilege = Access::Privilege::kOperate; // TODO: get actual request privilege
+        err                                = Access::GetAccessControl().Check(subjectDescriptor, requestPath, requestPrivilege);
+        if (err != CHIP_NO_ERROR)
+        {
+            const ConcreteCommandPath concretePath(endpointId, clusterId, commandId);
```

Comment:
We're now creating this same `ConcreteCommandPath` object in four places: here, for the call to CommandExists, and for the calls to `MatterPreCommandReceivedCallback` and `DispatchCommand`.

We should really create it once.  I know you didn't do the three (why??) existing callsites, but please don't add another one.  Just move this declaration to right above the `CommandExists` call and use it in all four places.

Reply:
I started doing that but wanted to keep this PR simple. I'm happy to refactor it to one place, that's how it ultimately will be.

Reply:
I should mention the reason I didn't keep the refactor of the ConcreteCommandPath, is if you look carefully, you'll see there are cases where we error out and return it as status, before we've fully constructed it...

Reply:
I made separate PR #12684 for just this refactor of the concrete command path.